pyth-sdk-solana/src/state.rs

```diff
@@ -410,14 +424,14 @@ impl PriceAccount {
     }
 }
 
-fn load<T: Pod>(data: &[u8]) -> Result<&T, PodCastError> {
+fn load<T: Pod>(data: &[u8]) -> Result<&T, CheckedCastError> {
     let size = size_of::<T>();
     if data.len() >= size {
         Ok(from_bytes(cast_slice::<u8, u8>(try_cast_slice(
```

Comment:
Unrelated : what is the point of all these `cast_slice` and `try_cast_slice` seems redundant.